cellxgene_schema_cli/cellxgene_schema/validate.py

```diff
@@ -449,43 +449,25 @@ def _validate_feature_id(self, feature_id: str, df_name: str):
         return
 
     @staticmethod
-    def _chunk_matrix(
-        matrix: Union[np.ndarray, sparse.spmatrix],
-        obs_chunk_size: Optional[int] = 10_000,
-    ):
-        """
-        Iterator which chunks the _named_ or _specified_ matrix by the
-        first (obs) dimension
-
-        The parameter type restrictions are strictly for ensuring that the
-        AnnData read fast-path is used (as of AnnData 0.8.0).
-
-        Iterator produces a sequence of tuples, each containing
-        (chunk, start, end)
-        """
-        start = 0
-        n = matrix.shape[0]
-        for i in range(int(n // obs_chunk_size)):
-            logger.debug(f"_chunk_matrix [{i} of {math.ceil(n / obs_chunk_size)}]")
-            end = start + obs_chunk_size
-            yield (matrix[start:end], start, end)
-            start = end
-        if start < n:
-            yield (matrix[start:n], start, n)
-
-    def _count_matrix_nonzero(self, matrix_name: str, matrix: Union[np.ndarray, sparse.spmatrix]) -> int:
-        if matrix_name in self.number_non_zero:
-            return self.number_non_zero[matrix_name]
-
-        logger.debug(f"Counting non-zero values in {matrix_name}")
-
-        nnz = 0
-        matrix_format = get_matrix_format(self.adata, matrix)
-        for matrix_chunk, _, _ in self._chunk_matrix(matrix):
-            nnz += matrix_chunk.count_nonzero() if matrix_format != "dense" else np.count_nonzero(matrix_chunk)
-
-        self.number_non_zero[matrix_name] = nnz
-        return nnz
+    def count_matrix_nonzero(matrix: DaskArray, filter_by_column: pd.Series = None) -> int:
+        def count_nonzeros(
+            matrix_chunk: Union[np.ndarray, sparse.spmatrix], is_sparse_matrix: bool, filter_by_column: pd.Series
+        ) -> np.array:
+            if filter_by_column is not None:
```

Comment:
nit: you should move the filtering outside of count_nonzeros. This will reduce the amount of dataset sent between jobs. and should speed things up.